helm/prometheus/templates/configmap.yaml

```diff
@@ -7,7 +7,8 @@ metadata:
     heritage: {{ .Release.Service }}
     prometheus: {{ .Release.Name }}
     release: {{ .Release.Name }}
-  name: {{ template "fullname" . }}
+    role: alert-rules
```

Comment:
Could you explain why did you add this label role?

Reply:
when i added chart specific prefixes to the template names to prevent overlapping, it broke the rules discovery powered by https://github.com/coreos/prometheus-operator/blob/master/helm/prometheus/templates/prometheus.yaml#L72
```yaml
ruleSelector:
    matchLabels:
      app: {{ template "name" . }}
      prometheus: {{ .Release.Name }}
```
because in order for this selector to discover rules declared in downstream charts, they are required to rely on the "name" template name to resolve the same as in the parent chart.

One solution would have been to remove the app label alltogether, but the selector would've then matched all the configmaps declared in the chart, amongst which some are not used to define alerting rules.

another one would have been to allow for custom labels to be defined on alerts-defining configmaps to provide a way for the user to make them match with the selector